src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -389,6 +554,32 @@ ParsedQuery Visitor::visitTypesafe(Parser::SelectQueryContext* ctx) {
                        ctx->getText());
 }
 
+// ____________________________________________________________________________________
+GraphPatternOperation::Subquery Visitor::visitTypesafe(
+    Parser::SubSelectContext* ctx) {
+  ParsedQuery query;
+  visibleVariables_.emplace_back();
+  query._clause = visitTypesafe(ctx->selectClause());
+  query._rootGraphPattern = visitTypesafe(ctx->whereClause());
+  query.setNumInternalVariables(numInternalVariables_);
+  query.addSolutionModifiers(visitTypesafe(ctx->solutionModifier()));
+  numInternalVariables_ = query.getNumInternalVariables();
+  auto values = visitTypesafe(ctx->valuesClause());
+  if (values.has_value()) {
+    query._rootGraphPattern._children.emplace_back(std::move(values.value()));
+  }
+  // Process Variables that are visible by children of this query.
+  for (const auto& variable : visibleVariables_.back()) {
+    query.registerVariableVisibleInQueryBody(variable);
+  }
+  visibleVariables_.pop_back();
+  // Variables that are selected in this query are visible in the parent query.
+  for (const auto& variable : query.selectClause().getSelectedVariables()) {
+    visibleVariables_.back().emplace_back(variable);
```

Comment:
Is it clear on all paths that there is still a vector left in the `visibleVariables` after the `popBack` (e.g. also when directly calling this for a test?)

Reply:
Now it is yes. `visibleVariables_` is initialized with one sub vector. Only `whereClauseContext` modifies it. It append a new vector, visit its child clauses and then removes the vector again. All other visit functions only append to the last inner vector.